src/main/java/org/stellar/sdk/MemoId.java

```diff
@@ -10,9 +10,6 @@
   private long id;
 
   public MemoId(long id) {
-    if (id < 0) {
-      throw new IllegalArgumentException("id must be a positive number");
-    }
```

Comment:
Please revert this change. Memo [must](https://github.com/stellar/stellar-core/blob/master/src/xdr/Stellar-transaction.x#L291) be a positive number (or zero).